Tasks/VsTest/vstest.ts

```diff
@@ -343,8 +350,249 @@ function getTestResultsDirectory(settingsFile: string, defaultResultsDirectory:
+                            dataCollectorArray.push({});
+                            result.TestSettings.Execution[0].AgentRule[0].DataCollectors[0].DataCollector = dataCollectorArray;
+                            dataCollectorNode = dataCollectorArray[dataCollectorArray.length - 1];
+                        }
+                    }
+                }
+                if (dataCollectorNode) {
+                    tl.debug("Setting attributes for test impact data collector");
+                    dataCollectorNode.$ = getTestImpactAttributes(vsVersion);
+                }
+                var builder = new xml2js.Builder();
+                var runSettingsForTestImpact = builder.buildObject(result);
+                saveToFile(runSettingsForTestImpact, ".testsettings")
+                    .then(function (fileName) {
+                    cleanUp(settingsFile);
+                    defer.resolve(fileName);
+                    return defer.promise;
+                })
+                    .fail(function (err) {
+                    tl.debug(exitErrorMessage);
+                    tl.warning(err);
+                    defer.resolve(settingsFile);
+                });
+            });
+        })
+        .fail(function (err) {
+            tl.warning(err);
+            tl.debug(exitErrorMessage);
+            defer.resolve(settingsFile);
+        });
+    return defer.promise;
+}
+
+
+function createRunSettingsForTestImpact(vsVersion: number, settingsFile: string, exitErrorMessage: string): Q.Promise<string> {
+    var defer = Q.defer<string>();
+    tl.debug("No settings file provided or the provided settings file does not exist. Creating run settings file for enabling test impact data collector.");
+    var runSettingsForTIA = '<?xml version="1.0" encoding="utf-8"?><RunSettings><DataCollectionRunSettings><DataCollectors>' +
+        '<DataCollector uri="' + getTICollectorURI() + '" ' +
+        'assemblyQualifiedName="' +  getTIAssemblyQualifiedName(vsVersion) + '" ' +
+        'friendlyName="' + getTIFriendlyName() + '"/>' +
+        '</DataCollectors></DataCollectionRunSettings></RunSettings>';
+    saveToFile(runSettingsForTIA, ".runsettings")
+        .then(function (fileName) {
+            defer.resolve(fileName);
+            return defer.promise;
+        })
+        .fail(function (err) {
+            tl.debug(exitErrorMessage);
+            tl.warning(err);
+            defer.resolve(settingsFile);
+        });
+    return defer.promise;
+}
+
+function setupRunSettingsFileForTestImpact(vsVersion: number, settingsFile: string): Q.Promise<string> {
+    var defer = Q.defer<string>();
+    var tiaEnabled = tl.getVariable('tia.enabled');
+    var exitErrorMessage = "Error occured while setting in test impact data collector. Continuing...";
+    if (tiaEnabled && tiaEnabled == "true") {
+        if (settingsFile && settingsFile.split('.').pop().toLowerCase() == "testsettings") {
```

Comment:
if(tiaEnabled)
tiaEnabled type is string
So it evaluates to TRUE even if user has set the variable as false.